index.bs

```diff
@@ -1939,48 +1965,71 @@ To <dfn>parse summary buckets</dfn> given a [=map=] |map| and an integer |maxEve
-1. [=list/iterate|For each=] |item| of |map|["`trigger_specs`"]:
-    1. If |item| is not a [=map=], return an error.
+|sourceTime|, a [=source type=] |sourceType|, a [=duration=] |expiry|, and a
+[=trigger-data matching mode=] |matchingMode|:
+
+1. Let |defaultReportWindows| be the result of
+    [=parsing top-level report windows=] with |map|, |sourceTime|, |sourceType|,
+    and |expiry|.
+1. If |defaultReportWindows| is an error, return an error.
+1. Let |specs| be a new [=trigger spec map=].
+1. If [=experimental Flexible Event support=] is true and
+    |map|["`trigger_specs`"] [=map/exists=]:
+    1. If |map|["`trigger_data`"] [=map/exists=], return an error.
+    1. If |map|["`trigger_specs`"] is not a [=list=] or its
+        [=list/size=] is greater than [=max distinct trigger data per source=],
+        return an error.
+    1. [=list/iterate|For each=] |item| of |map|["`trigger_specs`"]:
+        1. If |item| is not a [=map=], return an error.
+        1. Let |spec| be a new [=trigger spec=] with the following items:
+            : [=trigger spec/event-level report windows=]
+            :: |defaultReportWindows|
+        1. If |item|["`event_report_windows`"] [=map/exists=]:
+            1. Let |reportWindows| be the result of
+                [=parsing report windows=] with |item|["`event_report_windows`"],
+                |sourceTime|, and |expiry|.
+            1. If |reportWindows| is an error, return it.
+            1. Set |spec|'s [=trigger spec/event-level report windows=] to
+                |reportWindows|.
+        1. If |item|["`trigger_data`"] does not [=map/exist=], return an error.
+        1. Let |allowEmpty| be false.
+        1. If the result of running
+            [=parse trigger data into a trigger spec map=] with
+            |item|["`trigger_data`"], |spec|, |specs|, and |allowEmpty| is
+            false, return an error.
+1. Otherwise:
     1. Let |spec| be a new [=trigger spec=] with the following items:
         : [=trigger spec/event-level report windows=]
         :: |defaultReportWindows|
-    1. If |item|["`event_report_windows`"] [=map/exists=]:
-        1. Let |reportWindows| be the result of
-            [=parsing report windows=] with |item|["`event_report_windows`"],
-            |sourceTime|, and |expiry|.
-        1. If |reportWindows| is an error, return it.
-        1. Set |spec|'s [=trigger spec/event-level report windows=] to
-            |reportWindows|.
-    1. If |item|["`trigger_data`"] does not [=map/exist=], is not a [=list=], or
-        [=list/is empty=], or its [=list/size=] is greater than
-        [=max distinct trigger data per source=], return an error.
-    1. [=list/iterate|For each=] |triggerData| of |item|["`trigger_data`"]:
-        1. If |triggerData| is not an integer or cannot be represented by an
-            unsigned 32-bit integer, or |specs|[|triggerData|] [=map/exists=],
+    1. If |map|["`trigger_data`"] [=map/exists=]:
+        1. Let |allowEmpty| be true.
```

Comment:
I'm still not completely clear on the need for different treatment of the field, `trigger_data`, with the same name in the trigger specs offered in a registration.

Reply:
If top-level `trigger_data` cannot be empty, there is no way to cause a source registration to have an empty set of trigger specs, which *is* allowed in Full Flex.

Reply:
In effect, the spec is taking a `List<uint32>` and transforming it into a `Map<uint32, Spec>`, where a `Spec` contains `event report windows`. (In Full Flex, `Spec` also contains `summary_buckets`.)

The spec doesn't actually do any transofmration between *JSON* representations. But here are equivalencies between this PR and hypothetical Full Flex:

```jsonc
// empty example
{"trigger_data": []}
{"trigger_specs": []}
```
```jsonc
// non-empty example
{"trigger_data": [0, 1]}
{"trigger_specs": [{"trigger_data": [0, 1]}]
{"trigger_specs": [{"trigger_data": [0]}, {"trigger_data": [1]}]}
```

Reply:
> If top-level `trigger_data` cannot be empty, there is no way to cause a source registration to have an empty set of trigger specs, which _is_ allowed in Full Flex.

This isn't full-flex. So far, full flex is the `trigger_specs` field, as far as I'm aware.